Servers/Administration/OJS.Servers.Administration/Controllers/ProblemsController.cs

```diff
@@ -607,17 +645,41 @@ private static void AddSubmissionTypes(Problem problem, AdminActionContext actio
     {
         var newSubmissionTypes = actionContext.GetSubmissionTypes()
             .Where(x => x.IsChecked)
-            .Select(x => new SubmissionTypeInProblem
+            .Select(x =>
             {
-                ProblemId = problem.Id,
-                SubmissionTypeId = int.Parse(x.Value!.ToString() !),
-                SolutionSkeleton = x.Expand != null && x.Expand.Value != null
-                    ? x.Expand.Value!.ToString() !.Compress()
-                    : Array.Empty<byte>(),
+                var psted = new ProblemSubmissionTypeExecutionDetails();
+                psted.ProblemId = problem.Id;
+                psted.SubmissionTypeId = int.Parse(x.Value!.ToString() !);
+                if (x.Expand == null)
+                {
+                    return psted;
+                }
+
+                psted.SolutionSkeleton = x.Expand.First(y =>
+                    y.Name == x.Name + " " + AdditionalFormFields.SolutionSkeletonRaw).Value != null
+                    ? x.Expand.First(y => y.Name == x.Name + " " + AdditionalFormFields.SolutionSkeletonRaw)
+                        .Value!.ToString() !.Compress()
+                    : Array.Empty<byte>();
+
+                var timeLimitValue = x.Expand.First(y =>
+                    y.Name == x.Name + " " + AdditionalFormFields.TimeLimit).Value;
```

Comment:
This logic for generating the Name is repeated. Better to be extracted.

Reply:
Done.